pkg/api/pod/util_test.go

```diff
@@ -1754,3 +1754,91 @@ func TestDropSubPathExpr(t *testing.T) {
 		}
 	}
 }
+
+func TestDropTopologySpreadConstraints(t *testing.T) {
+	makePodWithTopologySpreadConstraints := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				TopologySpreadConstraints: []api.TopologySpreadConstraint{
+					{
+						MaxSkew:           1,
+						TopologyKey:       "k8s.io/hostname",
+						WhenUnsatisfiable: api.DoNotSchedule,
+					},
+				},
+			},
+		}
+	}
+	makePodWithoutTopologySpreadConstraints := func() *api.Pod {
+		return &api.Pod{}
+	}
+
+	podInfo := []struct {
+		description string
+		makePod     func() *api.Pod
+	}{
+		{
+			description: "has topology spread constraints",
+			makePod:     makePodWithTopologySpreadConstraints,
+		},
+		{
+			description: "does not have topology spread constraints",
+			makePod:     makePodWithoutTopologySpreadConstraints,
+		},
+		{
+			description: "is nil",
+			makePod:     func() *api.Pod { return nil },
+		},
+	}
+
+	for _, enabled := range []bool{true, false} {
```

Comment:
This set of nested for loops is MUCH harder to understand than just putting all combos in the table with expected results. Tests with a lot of logic are much harder to verify.

Even if the rest of the file is like this, I would prefer this test to change.

Also, does this need to be tested on things that embed pod templates?

Reply:
> Also, does this need to be tested on things that embed pod templates?

The dropping of fields will apply to both pod and pod templates and ideally should be tested in both places

Reply:
Done.